### PR TITLE
[Single Span Sampling] Rule configuration settings

### PR DESCRIPTION
**Pull request by marcotc**
Follow up from #2098 

This PR adds `ddtrace` configuration options for Single Span Sampling.

This configuration can come either directly from an environment variable, or from from a file. If both are provided, the environment variable value prevails.

The relevant spec section is here: https://github.com/DataDog/architecture/blob/master/rfcs/apm/integrations/single-span-ingestion-control/README.md#recommended-solution-span-sampling-rules

This PR is not responsible for consuming the values, just exposing the settings and documenting them.

A few TODO are left as placeholders to URLs for the complete rule sampling documentation.

### CODE REVIEW

File: lib/datadog/core/configuration/settings.rb [hunk 1]
```diff
@@ -553,6 +554,47 @@ def initialize(*_)
...
+                        'also providing DD_SPAN_SAMPLING_RULES as their configuration conflicts.'
+                      )
+                    end
+                    rules
+                  elsif rules_file
+                    begin
+                      File.read(rules_file)
+                    rescue => e
+                      # `File#read` errors have clear and actionable messages, no need to add extra exception info.
+                      Datadog.logger.warn("Cannot read span sampling rules file: #{e.message}")
```
Comment (ivoanjo) [inline lib/datadog/core/configuration/settings.rb:590]: Should we also mention in this message:

* :one: that we're trying to read what came from `DD_SPAN_SAMPLING_RULES_FILE` and its value
* :two: what's the expected outcome of this failure. E.g. no span sampling rules will be applied.

File: docs/GettingStarted.md [hunk 2]
```diff
@@ -2033,6 +2034,7 @@ end
...
 | `tracing.sampling.default_rate`                         | `DD_TRACE_SAMPLE_RATE`         | `nil`                                                             | Sets the trace sampling rate between `0.0` (0%) and `1.0` (100%). See [Application-side sampling](#application-side-sampling) for details.                                                                                                  |
 | `tracing.sampling.rate_limit`                           | `DD_TRACE_RATE_LIMIT`          | `100` (per second)                                                | Sets a maximum number of traces per second to sample. Set a rate limit to avoid the ingestion volume overages in the case of traffic spikes.                                                                    |
+| `tracing.sampling.span.rules`                           | `DD_SPAN_SAMPLING_RULES`,`ENV_SPAN_SAMPLING_RULES_FILE` | `nil`                                    | Sets [Single Span Sampling](#single-span-sampling) rules. These rules allow you to keep spans even when their respective traces are dropped.                                                                                              |
```
[the remark this answers is not in the document]
Comment (marcotc) [reply docs/GettingStarted.md:2037]: `DD_SPAN_SAMPLING_RULES` would possibly directly translate to `c.span_sampling_rules` or `c.span_sampling.rules`.
But we have product namespacing, so minimally `c.tracing.span_sampling_rules`.
We also already have sampling, which is somewhat reasonable, so `c.tracing.sampling.span_rules`.

I can live with that.

File: lib/datadog/core/configuration/settings.rb [hunk 3]
```diff
@@ -553,6 +554,47 @@ def initialize(*_)
...
+                  rules = ENV[Tracing::Configuration::Ext::Sampling::Span::ENV_SPAN_SAMPLING_RULES]
+                  rules_file = ENV[Tracing::Configuration::Ext::Sampling::Span::ENV_SPAN_SAMPLING_RULES_FILE]
+
+                  if rules
+                    if rules_file
+                      Datadog.logger.warn(
+                        'Both DD_SPAN_SAMPLING_RULES and DD_SPAN_SAMPLING_RULES_FILE were provided: only ' \
+                        'DD_SPAN_SAMPLING_RULES will be used. Please do not provide DD_SPAN_SAMPLING_RULES_FILE when ' \
+                        'also providing DD_SPAN_SAMPLING_RULES as their configuration conflicts.'
+                      )
```
[the remark this answers is not in the document]
Comment (marcotc) [reply lib/datadog/core/configuration/settings.rb]: Done!

File: docs/GettingStarted.md [hunk 4]
```diff
@@ -2177,6 +2179,12 @@ trace.reject!
 trace.keep!
 ```
 
+#### Single Span Sampling
+
+You can configure sampling rule that allow you keep spans despite their respective traces being dropped by a trace-level sampling rule.
+
+See (TODO: Insert documentation URL here when published) for the full documentation on Single Span Sampling.
```
[the remark this answers is not in the document]
Comment (marcotc) [reply docs/GettingStarted.md]: Captured it in the main feature branch PR: https://github.com/DataDog/dd-trace-rb/pull/2128

**Pull request merged**